File: 2582-pass-the-pillow/2582-pass-the-pillow.cpp
class Solution {
public:
    int passThePillow(int n, int time) {
        int cycle_length = 2 * n - 2;
        int effective_time = time % cycle_length;
        
        if (effective_time < n) {
            return effective_time + 1;
        } else {
            return 2 * n - effective_time - 1;
        }
    }
};

File: 2582-pass-the-pillow/2582-pass-the-pillow.cs
public class Solution {
    public int PassThePillow(int n, int time) {
        int position = 1; // Starting position
        bool forward = true; // Initial direction is forward

        for (int t = 1; t <= time; ++t) {
            if (forward) {
                // Move forward
                if (position == n) {
                    // Change direction to backward if at the last person
                    forward = false;
                    position--;
                } else {
                    position++;
                }
            } else {
                // Move backward
                if (position == 1) {
                    // Change direction to forward if at the first person
                    forward = true;
                    position++;
                } else {
                    position--;
                }
            }
        }

        return position;
    }
}